### ISSUE TITLE
Flying units can land on top of buildings

### ISSUE THREAD

**Issue by leveleditor** [leveleditor/CrystAlien-Conflict-Flash-Wrapper#3]
Found this in my notes, so I'm adding it here to keep track of it for later.

In the last Astro Special Ops level, if you place too many units around your base, and then build a switch fighter, it will hover over and then land on the HyperSonic spacecraft.
http://marsmissionwiki.wikifoundry.com/photo/15423496/Bug+in+%22Jacks%22+Level

This is because flying units can't normally go where a building is (can't click to send them there), but when they do go there, or are built there due to running out of available space, they only see the traversable land tile, so they land there. Adding a check for buildings before landing should fix this.

Basically, buildings only have a single blocked tile, allowing units to travel through larger buildings, but if a flying unit happens to rest over one of the tiles that allow travel, they go ahead and land there. This isn't game breaking, but can be annoying if you expected the unit to be flying and it gets attacked by a ground unit. Also the units are harder to select when they have landed inside a building, as the building blocks the click selection, so you have to use the click and drag area selection.
